proto/pocket/shared/service.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+package pocket.shared;
+
+// NOTE that the `shared` package is not a Cosmos module,
+// but rather a manually created package to resolve circular dependencies.
+
+option go_package = "pocket/x/shared/types";
+
+// Service is the main service message describing all configurations for a service
+// TODO_RESEARCH(@olshansk): Should these be permissioned or permissionless?
+message Service {
+    ServiceId id = 1; // Unique and semantic identifier for the service
+    ServiceConfig config = 2; // Configuration settings for the service
```

Comment:
If the `Service` is a standalone struct. I feel like `ServiceConfig` should not contain `Endpoint`s, which is more of a `Supplier` specific thing. It should be enough to have `RPCType`s there which describe the supported transports of that service.

I would suggest we either replace `ServiceConfig config` with `repated RPCType rpc_types` or if we chose to keep `ServiceConfig`, this later should contain the `repeated RPCType rpc_types` entry instead of `repeated Endpoint endpoints`.

If anyway we want to describe `Endpoints` . It would be better to have a `SupplierService` which will be in terms of `Service` and `Supplier` but we still need the standalone variant describing a service and its possible RPC types
```proto
message Service {
  ServiceId id = 1;
  repeated RPCType rpc_types = 2;
}
```
and optionnaly
```proto
message SupplierService {
  Service service = 1;
  ServiceConfig config = 2; // config.enpoints[].rpc_type must be a subset of Service.rpc_types
}
```

Reply:
@red-0ne PTAL at the changes and LMK WYT